src/middlewared/middlewared/plugins/network.py

```diff
@@ -2735,40 +2735,26 @@ async def configure_http_proxy(middleware, *args, **kwargs):
 
 
 async def attach_interface(middleware, iface):
-    if not iface:
-        return
-
-    # We dont handle the following interfaces in middlewared
-    if iface.startswith(('epair', 'tun', 'tap', 'veth', 'kube-bridge')):
-        return
-
-    iface = await middleware.call('interface.query', [('name', '=', iface)])
-    if not iface:
-        return
-
-    iface = iface[0]
-    # We only want to sync physical interfaces that are hot-plugged,
-    # not cloned interfaces with might be a race condition with original devd.
-    # See #33294 as an example.
-    if iface['state']['cloned']:
```

Comment:
Would this be `false` for a cloned interface ?

Reply:
Yes it would be `false`. As a matter of fact, anything in `CLONED_PREFIXES` is tagged as `iface['state']['cloned']`. However, we're calling `interface.query()` before the bond is fully setup which leads to the race condition that I mention in the description.

Reply:
Exactly my point, we don't need to have this complete diff. We should fix the root cause which is `interface.query` failing.

Reply:
Why though? This avoids calling `interface.query` entirely since we don't even care about these interfaces in the first place?

Reply:
And fixing `interface.query` is just suppressing a known failure scenario. It's not fixing the root cause by any means.

Reply:
To be sure i follow, are you suggesting that we should not fix `interface.query` and have a traceback ? Which means regardless of the changes any client consuming `interface.query` would error out including UI ?
Also i am not i follow the root cause in this case, isn't it `interface.query` because it's not reporting the protocol correctly as it's still being set and errors out ? From my understanding, we are just fixing a side effect of the original problem and not the problem itself ?
I think the best way forward would be able to report that the protocol is not set yet in `interface.query` and that should cover the current diff as well and any other client consuming this endpoint won't error out as well.

Reply:
You're right, if we have a 3rd party API consumer calling `interface.query` timed properly then it will still traceback if another consumer (i.e. webUI) is setting up a bond interface.

The changes here fix the traceback just from a webUI consumer standpoint since it triggers a `udevd` hook. So I'll ask you 3 follow-up questions.

1. Would you want the `protocol` key in `__getstate__()` to return null or an empty string?
2. Keeping the current diff still fixes the race condition that causes the `NoneType` specifically with the `udevd_ifnet_hook` function so, imo, I believe the fix is still correct would you agree?
3. if you agree to question number 2, how about I change the `__gestate__()` method as described in this PR too?

Reply:
For (1), i think we can return `None` and perhaps another flag `protocol_set` which can be a boolean value of `protocol`.
I don't follow why we would need (2) if we fix (1) ?